internal/services/keyvault/key_vault_resource.go

```diff
@@ -639,33 +632,46 @@ func resourceKeyVaultUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
 
-	if d.HasChange("contact") {
-		contacts := dataplane.Contacts{
-			ContactList: expandKeyVaultCertificateContactList(d.Get("contact").(*pluginsdk.Set).List()),
-		}
+	if !features.FivePointOh() {
+		if d.HasChange("contact") {
+			contacts := dataplane.Contacts{
+				ContactList: pointer.To(make([]dataplane.Contact, 0)),
+			}
 
-		vaultUri := ""
-		if existing.Model != nil && existing.Model.Properties.VaultUri != nil {
-			vaultUri = *existing.Model.Properties.VaultUri
-		}
+			if len(d.Get("contact").(*pluginsdk.Set).List()) > 0 || d.Get("contact").(*pluginsdk.Set).List()[0] != nil {
+				for _, item := range d.Get("contact").(*pluginsdk.Set).List() {
+					v := item.(map[string]interface{})
+					contacts.ContactList = pointer.To(append(*contacts.ContactList, dataplane.Contact{
+						Name:         pointer.To(v["name"].(string)),
+						EmailAddress: pointer.To(v["email"].(string)),
+						Phone:        pointer.To(v["phone"].(string)),
+					}))
+				}
+			}
 
-		if vaultUri == "" {
-			return fmt.Errorf("failed to get vault base url for %s: %s", *id, err)
-		}
+			vaultUri := ""
+			if existing.Model != nil && existing.Model.Properties.VaultUri != nil {
+				vaultUri = *existing.Model.Properties.VaultUri
+			}
```

Comment:
I appreciate this was existing code, but while we're here.... 😄 

```suggestion
			if existing.Model != nil && existing.Model.Properties != nil {
				vaultUri = pointer.From(existing.Model.Properties.VaultUri)
			}
```

Reply:
Updated